sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -73,7 +73,12 @@ private[sql] class JacksonGenerator(
 
   private val gen = {
     val generator = new JsonFactory().createGenerator(writer).setRootValueSeparator(null)
-    if (options.pretty) generator.useDefaultPrettyPrinter() else generator
+    val ppGenerator = if (options.pretty) generator.useDefaultPrettyPrinter() else generator
+    if (options.writeNonAsciiCharacterAsCodePoint) {
+      generator.setHighestNonEscapedChar(0x7F)
```

Comment:
This code means that we cannot set both `options.pretty` and `options.writeNonAsciiCharacterAsCodePoint`. If this is true, shall we document it somewhere?